components/prize_pool/commons/prize_pool_import.lua

```diff
@@ -0,0 +1,512 @@
+		)
+	end
+
+	return placements
+end
+
+function Import._emptyPlacement(priorPlacement, placementSize)
+	priorPlacement = priorPlacement or {}
+	local placeStart = (priorPlacement.placeEnd or 0) + 1
+	local placeEnd = (priorPlacement.placeEnd or 0) + placementSize
+
+	return Placement(
+		{placeStart = placeStart, placeEnd = placeEnd},
+		priorPlacement.parent,
+		priorPlacement.placeEnd or 0
+	)
+end
+
+function Import._mergePlacement(lpdbEntries, placement)
+	for opponentIndex, opponent in ipairs(lpdbEntries) do
+		placement.opponents[opponentIndex] = Import._mergeEntry(
+			opponent,
+			Table.mergeInto(placement:_parseOpponents{{}}[1], placement.opponents[opponentIndex]),
+			placement
+		)
+	end
+
+	assert(
+		#placement.opponents <= 1 + placement.placeEnd - placement.placeStart,
+		'Import: Too many opponents returned from query for placement range '
+			.. placement:_displayPlace():gsub('&#045;', '-')
+	)
+
+	return placement
+end
+
+function Import._mergeEntry(lpdbEntry, entry, placement)
+	if
+		not Opponent.isTbd(entry.opponentData) -- valid manual input
+		or Table.isEmpty(lpdbEntry.opponent) -- irrelevant lpdbEntry
+		or Opponent.isTbd(lpdbEntry.opponent)
+	then
+		return entry
+	end
+
+	return Table.deepMergeInto(Import._entryToOpponent(lpdbEntry, placement), Import._removeTbd(entry))
+end
+
+function Import._removeTbd(entry)
+	-- if we get here we know entry is a TBD Opponent
+	-- but it still might hold data (e.g. flag or race of players) for non party opponents
+	local opponent = entry.opponentData
+	if not Opponent.typeIsParty(opponent.type) then
+		-- TBD opponents of non party types do not hold additional data
+		entry.opponentData = {}
+	else
+		opponent.type = nil
+		for _, player in pairs(opponent.players or {}) do
+			if Opponent.playerIsTbd(player) then
+				player.displayName = nil
+				player.pageIsResolved = nil
+				player.pageName = nil
+			end
+		end
+	end
+
+	return entry
+end
+
+function Import._entryToOpponent(lpdbEntry, placement)
+	local additionalData = {}
+	if lpdbEntry.needsLastVs then
+		additionalData = Import._groupLastVsAdditionalData(lpdbEntry)
+	end
+
+	local score = additionalData.score or Import._getScore(lpdbEntry.opponent)
+	local vsScore = additionalData.vsScore or Import._getScore(lpdbEntry.vsOpponent)
+	local lastVsScore
+	if score or vsScore then
+		lastVsScore = (score or '') .. '-' .. (vsScore or '')
+	end
+
+	return placement:_parseOpponents{{
+		Import._removeIfTbd(lpdbEntry.opponent),
+		wdl = (not lpdbEntry.needsLastVs) and Import._makeGroupScore(lpdbEntry) or nil,
+		lastvs = additionalData.lastVs or Import._removeIfTbd(lpdbEntry.vsOpponent),
+		lastvsscore = additionalData.score or lastVsScore,
+		date = lpdbEntry.date,
+		isAlreadyParsed = true,
+	}}[1]
+end
+
+function Import._removeIfTbd(opponent)
+	return (Table.isEmpty(opponent) or Opponent.isTbd(opponent))
+		and {} or Table.merge(opponent, {isAlreadyParsed = true})
+end
+
+function Import._makeGroupScore(lpdbEntry)
```

Comment:
```suggestion
function Import._formatGroupScore(lpdbEntry)
```

Reply:
will do when on pc in the evening (searching for the call of it is pretty shit on phone)